### PR TITLE
FORPRDEV-18387: Update website for 23.0.1 release

### PR DESCRIPTION
**Pull request by RichardBuntLinaro**

### CODE REVIEW

File: forge-website/content/JsonFiles/open_source.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [
     {
-        "version": "23.0",
+        "version": "23.0.1",
```
Comment (joscot-linaro) [inline forge-website/content/JsonFiles/open_source.json:3]: We have made the change to include more historical releases for our open-source links. Do you think it is worth adding another entry for this or would it be okay to just keep entries per Major?

**Review by joscot-linaro — COMMENTED** [review]
Also missing some documentation link updates in:
- forge-website/src/pages/documentation.jsx
- forge-website/src/pages/downloadForge/index.jsx
- forge-website/src/pages/licenseServer.jsx

File: forge-website/content/JsonFiles/open_source.json (hunk 1 above)
Comment (RichardBuntLinaro) [reply forge-website/content/JsonFiles/open_source.json:3]: Added a new entry for 23.0.1. We are required to provide the source for all versions available for download.

**Review by joscot-linaro — APPROVED** [review]
LGTM

**Pull request merged**